### PR TITLE
[jobs] use node-e2e-project boskos pool for ci-containerd-node-e2e

### PR DESCRIPTION
**Pull request by hasheddan**
Trying same change made in https://github.com/kubernetes/test-infra/pull/17349

Ref: https://github.com/kubernetes/kubernetes/issues/89847

Signed-off-by: hasheddan <georgedanielmangum@gmail.com>

### CODE REVIEW

File: config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml [hunk 1]
```diff
@@ -101,7 +101,7 @@ presubmits:
         - --scenario=kubernetes_e2e
         - --
         - --deployment=node
-        - --gcp-project=k8s-c8d-pr-node-e2e
```
Comment (spiffxp) [inline config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml:104]: We don't have enough capacity in this pool to handle presubmits, can you try with a ci- job that uses this project first?

Comment (hasheddan) [reply config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml:104]: @spiffxp hmm I don't see any other jobs that use this specific project

Comment (hasheddan) [reply config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml:104]: We could try this one in `cri-containerd-node-e2e` https://testgrid.k8s.io/sig-node-containerd#containerd-node-conformance

**Pull request merged**